tests/integration/test_amendment_chain.py

```diff
@@ -0,0 +1,294 @@
+        return result
+
+    def _results(self, qry):
+        response = self._response(qry)
+        return response['results']
+
+    def test_simple_first_filing(self):
+        expected_filing = self.STOCK_FIRST_F1
+        self.create_filing(1, expected_filing)
+
+        # Refresh downstream `ofec_amendments_mv` and `ofec_filings_all_mv`
+        manage.refresh_materialized(concurrent=False)
+
+        # FilingsList view
+        # /filings/ endpoint
+
+        results = self._results(rest.api.url_for(FilingsList, committee_id=expected_filing['committee_id']))
+        assert len(results) == 1
+        list_result = results[0]
+
+        self.assert_filings_equal(list_result, expected_filing)
+
+        # FilingsView view
+        # /committee/<committee_id>/filings/ and /candidate/<candidate_id>/filings/
+
+        # TODO: Keep this test and then combine FilingsView and FilingsList
+        results = self._results(rest.api.url_for(FilingsView, committee_id=expected_filing['committee_id']))
+        assert len(results) == 1
+        view_result = results[0]
+
+        self.assert_filings_equal(view_result, expected_filing)
+
+    def test_simple_amended_filing(self):
+        expected_filing = self.STOCK_SECOND_F1
+
+        self.create_filing(1, self.STOCK_FIRST_F1)
+        self.create_filing(2, expected_filing)
+
+        # Refresh downstream `ofec_amendments_mv` and `ofec_filings_all_mv`
+        manage.refresh_materialized(concurrent=False)
+
+        # /filings/ endpoint
+
+        results = self._results(rest.api.url_for(FilingsList, committee_id=expected_filing['committee_id'], most_recent=True))
+        assert len(results) == 1
+        list_result = results[0]
+
+        self.assert_filings_equal(list_result, expected_filing)
+
+        # FilingsView view
+        # /committee/<committee_id>/filings/ and /candidate/<candidate_id>/filings/
+
+        results = self._results(rest.api.url_for(FilingsView, committee_id=expected_filing['committee_id'], most_recent=True))
+        assert len(results) == 1
+        view_result = results[0]
+
+        self.assert_filings_equal(view_result, expected_filing)
+
+    def test_multiple_form_3s(self):
+        form_3_q2_new = {
+            'committee_id': 'C006',
+            'report_year': 2018,
+            'amendment_indicator': 'N',
+            'receipt_date': datetime.date(2018, 7, 15),
+            'file_number': 20001,
+            'amendment_chain': [20001],
+            'previous_file_number': 20001,
+            'most_recent_file_number': 20003,
+            'most_recent': False,
+            'report_type': 'Q2',
+            'form_type': 'F3'
+        }
+        form_3_q2_amend_1 = {
+            'committee_id': 'C006',
+            'report_year': 2018,
+            'amendment_indicator': 'A',
+            'receipt_date': datetime.date(2018, 8, 15),
+            'file_number': 20002,
+            'amendment_chain': [20001, 20002],
```

Comment:
As we are mimicking the `efiling_amendment_chain_vw` logic, this field will have the entire longest chain, not the _longest chain so far_. This is different from the current behavior.

Reply:
@vrajmohan I didn't find that to be the case - check out `1170005` here: https://fec-dev-api.app.cloud.gov/v1/committee/C00330894/filings/?api_key=5yyI90SU3Xb73TVlv4wrEhQxYcCwMWCywQiGdYbJ&sort_hide_null=false&sort=-coverage_end_date&sort=report_type_full&sort=-beginning_image_number&per_page=100&page=1&form_type=F3&form_type=F3X&form_type=F3P&form_type=F3L&form_type=F4&form_type=F5&form_type=F7&form_type=F13&form_type=RFAI&report_type=-24&report_type=-48&request_type=-1&request_type=-5&request_type=-6&request_type=-9&cycle=2018

Reply:
Sorry if I misunderstood - do you mean, we should change the current behavior to show the longest chain so far?

Reply:
Looks like the front-end uses `amendment_chain` to generate the amendment number (which would also explain why it's wrong so often) https://github.com/fecgov/fec-cms/blob/develop/fec/fec/static/js/modules/helpers.js#L395-L418

Reply:
Actually there is a bigger question. When we force the most recent file number to be the highest file number for F1, F1M and F2, should we (do we need to) "fix" the `amendment_chain` as well? If yes, how?

Reply:
I think that we'd want to take the file ID's in descending order, and stop at the current file ID, if that's not too hard to implement.

Reply:
After discussing it with @PaulClark2, the plan forward is to make the F1/2/1M amendment chain logic match the existing F3 behavior for /filings/ so we don't have to change the front-end logic or crack open the F3 logic. I'll put in an issue to reconcile the /efile/filings amendment chain logic (goes all the way through the most recent filing) and the /filings/ amendment chain logic (stops at the filing you're looking at).